WordPress/Classes/Utility/ContextManager.swift

```diff
@@ -75,6 +83,27 @@ public class ContextManager: NSObject, CoreDataStack {
         }
     }
 
+    public func performAndSave<T>(_ block: @escaping (NSManagedObjectContext) throws -> T, completion: ((Result<T, Error>) -> Void)?) {
+        let context = newDerivedContext()
+        context.perform {
+            let result = Result(catching: { try block(context) })
+            if case .success = result {
+                self.saveContextAndWait(context)
+            }
+            DispatchQueue.main.async {
```

Comment:
Would it be useful for the queue on which this completion run be configurable by the caller?

Reply:
Ah! I saw that happens in the next commit